apps/api/src/server/applications/examination-timetable-items/examination-timetable-items.service.js

```diff
@@ -0,0 +1,105 @@
+import { eventClient } from '#infrastructure/event-client.js';
+import { NSIP_EXAM_TIMETABLE } from '#infrastructure/topics.js';
+import * as examinationTimetableRepository from '../../repositories/examination-timetable.repository.js';
+import * as examinationTimetableItemsRepository from '../../repositories/examination-timetable-items.repository.js';
+// @ts-ignore
+import { EventType } from '@pins/event-client';
+
+/**
+ * @typedef {import('../../../message-schemas/events/nsip-exam-timetable.js').NSIPExamTimetableItem} NSIPExamTimetableItem
+ * @typedef {import('../../../message-schemas/events/nsip-exam-timetable.js').NSIPExamTimetableItemDescriptionLineItem} NSIPExamTimetableItemDescriptionLineItem
+ */
+
+/**
+ * Grabs the description from the examinationTimetableItem and parses it into a description string and
+ * NSIPExamTimetableItemDescriptionLineItem array.
+ *
+ * @param {import('@prisma/client').ExaminationTimetableItem} examinationTimetableItem
+ * @returns { { description: string, eventLineItems: NSIPExamTimetableItemDescriptionLineItem[] } }
+ */
+function extractDescriptionAndLineItems(examinationTimetableItem) {
+	if (!examinationTimetableItem.description) {
+		return {
+			description: '',
+			eventLineItems: []
+		};
+	}
+
+	const parsedDescription = JSON.parse(examinationTimetableItem.description);
+
+	// The event line items are the bullet points that are added to the description. They are different from ExaminationTimetableItems.
+	const eventLineItems = parsedDescription.bulletPoints.map((/** @type {string} */ bulletPoint) => {
+		return {
+			eventLineItemDescription: bulletPoint
+		};
+	});
+
+	return {
+		description: parsedDescription.preText,
+		eventLineItems
+	};
+}
+
+/**
+ * Returns the payload containing all the examination timetable items.
+ *
+ * @param {import('@prisma/client').ExaminationTimetable} examinationTimetable
+ * @returns { Promise<NSIPExamTimetableItem[]> }
+ */
+async function buildExamTimetableItemsPayload(examinationTimetable) {
```

Comment:
@DavidJWhiteDev Once merged you will be able to use this same function to create the payload for unpublish

Reply:
Alternatively, you could branch off this branch and start using it right away:

`feature/boas-957-publish-exam-timetable-front-office`